// Generated from climate_comfort_service.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.climate_comfort_service;

// Namespace: comfort
// Climate comfort control with semantic operations

// Semantic comfort modes
enum comfort_mode_t {
  ECO = 0;
  COMFORT = 1;
  PERFORMANCE = 2;
  QUIET = 3;
}

// Air quality management modes
enum air_quality_mode_t {
  FRESH_AIR = 0;
  RECIRCULATE = 1;
  AUTO = 2;
  POLLUTION_PROTECT = 3;
}

// Zone configuration presets
enum zone_preset_t {
  ALL_ZONES = 0;
  DRIVER_ONLY = 1;
  FRONT_ONLY = 2;
  INDIVIDUAL = 3;
}

// High-level comfort request
message comfort_request_t {
  comfort_mode_t mode = 1;
  float target_temperature = 2;
  zone_preset_t zone_preset = 3;
  air_quality_mode_t air_quality_mode = 4;
}

// Current comfort status
message comfort_status_t {
  comfort_mode_t current_mode = 1;
  float average_cabin_temp = 2;
  bool achieving_target = 3;
  uint32 efficiency_score = 4;
  uint32 estimated_time_to_comfort = 5;
}

// Method: set_comfort
// Set cabin comfort using semantic preferences
message set_comfort_request {
  comfort_request_t request = 1;
}

message set_comfort_response {
  bool accepted = 1;
  uint32 estimated_time_seconds = 2;
  uint32 vss_signals_modified = 3;
}

service set_comfort_service {
  rpc set_comfort(set_comfort_request) returns (set_comfort_response);
}

// Method: quick_comfort_adjustment
// Quick comfort adjustments
message quick_comfort_adjustment_request {
  string adjustment = 1;
  uint32 intensity = 2;
}

message quick_comfort_adjustment_response {
  bool success = 1;
  string new_settings = 2;
}

service quick_comfort_adjustment_service {
  rpc quick_comfort_adjustment(quick_comfort_adjustment_request) returns (quick_comfort_adjustment_response);
}

// Method: optimize_for_efficiency
// Optimize current settings for efficiency
message optimize_for_efficiency_request {
  bool maintain_minimum_comfort = 1;
}

message optimize_for_efficiency_response {
  repeated string optimizations_applied = 1;
  uint32 efficiency_improvement_percent = 2;
}

service optimize_for_efficiency_service {
  rpc optimize_for_efficiency(optimize_for_efficiency_request) returns (optimize_for_efficiency_response);
}

// Method: get_comfort_status
// Get current comfort status
message get_comfort_status_request {
}

message get_comfort_status_response {
  comfort_status_t status = 1;
  repeated float zone_temperatures = 2;
}

service get_comfort_status_service {
  rpc get_comfort_status(get_comfort_status_request) returns (get_comfort_status_response);
}

// Method: prepare_for_occupancy
// Prepare cabin for upcoming occupancy
message prepare_for_occupancy_request {
  uint32 expected_occupancy_seconds = 1;
  comfort_request_t target_comfort = 2;
}

message prepare_for_occupancy_response {
  bool preparation_started = 1;
  bool ready_by_time = 2;
}

service prepare_for_occupancy_service {
  rpc prepare_for_occupancy(prepare_for_occupancy_request) returns (prepare_for_occupancy_response);
}


